src/components/InjectedComponents/PostDialog.tsx

```diff
@@ -279,7 +292,13 @@ export function PostDialog(props: PostDialogProps) {
     )
     const currentIdentity = or(props.currentIdentity, useCurrentIdentity())
     const [currentShareTarget, setCurrentShareTarget] = useState<(Profile | Group)[]>(() => [])
-
+    //#endregion
+    //#region Image Based Payload Switch
+    const [imagePayload, setImagePayload] = useState(useValueRef(steganographyModeSetting))
```

Comment:
Since it is per-post, it should not be a global setting IMO.

Reply:
I second that but,
https://github.com/DimensionDev/Maskbook/issues/923#issuecomment-610252980

Reply:
If we ignore the global one which will be useless totally. Since I use the global one as the default value here. It should be better with lazy initialization function `useState(() => steganographyModeSetting.value)`.

Reply:
In the perspective of usability, the Latest Chosen Value should be remembered in the lifecycle of software installation (made default for the next time, but not included in backups).

Reply:
@neruthes I think you are talking about global setting in another way. I will remove the global switch in the dashboard cause it's useless if we add per-post switch here.

Reply:
> @neruthes I think you are talking about global setting in another way. ~I will remove the global switch in the dashboard cause it's useless if we add per-post switch here.~ For now, the user can change this global setting in three places:
> 
> * the dropdown menu by clicking the Maskbook icon
> * the settings pane in the dashboard
> * the per-post composing dialog

Good, I got your point. I suggest that the option in the extension icon dropdown menu and the option in Settings page should be deleted as we are now making it a "step" in Post Composition.

Reply:
OK ;)

Reply:
It no longer a global setting flag. you may want to move it to another place, even make it a site-specific settings

Reply:
It's a network-specific setting right now.

Reply:
It's a network-specific setting right now.